src/object.rs

```diff
@@ -0,0 +1,321 @@
+use std::fmt;
+use std::collections::HashMap;
+use std::hash::{Hash,Hasher};
+use std::cell::RefCell;
+use std::rc::Rc;
+use ast;
+
+#[derive(Hash, Eq, PartialEq, Clone, Debug)]
+pub enum Object {
+    Int(i64),
+    Bool(bool),
+    String(String),
+    Return(Rc<Return>),
+    Function(Rc<Function>),
+    Builtin(Builtin),
+    Array(Rc<Array>),
+    Hash(Rc<MonkeyHash>),
+    Null,
```

Comment:
The Null object is referenced everywhere and I always create a new one when used. In the book a constant was used. I couldn't quite figure out how to do this in basic Rust. Would you use something like the `lazy_static` crate to define constant objects?